code/modules/mob/mob_helpers.dm

```diff
@@ -326,22 +326,17 @@ It's fairly easy to fix if dealing with single letters but not so much with comp
 #define PIXELS_PER_STRENGTH_VAL 16
 
 /proc/shake_camera(mob/M, duration, strength=1)
-	set waitfor = 0
-	if(!M || !M.client || M.shakecamera || M.stat || isEye(M) || isAI(M))
+	var/current_time = world.time
+	if(!M || !M.client || (M.shakecamera > current_time)|| M.stat || isEye(M) || isAI(M))
 		return
-	M.shakecamera = TRUE
+	M.shakecamera = current_time + max(TICKS_PER_RECOIL_ANIM, duration)
```

Comment:
The math I pasted earlier was correct. It delays twice for `TICKS_PER_RECOIL_ANIM` before and after the main loop, and delays `steps` times during the loop. Your current math is not accurate to the actual delay that the animation will take.

Reply:
Main loop is has a -1 at the end. It only runs if the duration is atleast 2x TICKS_PER_RECOIL_ANIM. So the duration account for the first animate and loops.

The last animate (the return to screen centre) can be safely overridded by the next screen shake (if there are any), it looks better when overridded (you don't want recoil to return perfectly to the centre before another screenshake)

Reply:
Alright fair enough.